packages/react-native-bpk-component-dialog/src/BpkDialog.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import memoize from 'lodash/memoize';
+import isNil from 'lodash/isNil';
+
+import {
+  commonPropTypes,
+  commonDefaultProps,
+  type CommonProps,
+} from './common-types';
+import NativeDialog from './NativeDialog';
+
+export type Props = {
+  ...$Exact<CommonProps>,
+};
+
+const createOnButtonClickHandler = memoize(actions => event => {
+  if (!isNil(event.nativeEvent.actionIndex)) {
+    const index = event.nativeEvent.actionIndex;
+    if (actions[index]) {
+      actions[index].callback();
+    }
+  }
+});
+
+const BpkDialog = (props: Props) => {
+  const { icon, ...rest } = props;
+
+  if (!icon.iconId.startsWith('bpk_')) {
```

Comment:
I wouldn't do this like. I think is better to use the `icons` export from bpk icons and those names so it's more familiar with RN folks, and then adjust the name here. 

E.g.:
```
import { icons } from `react-native-bpk-component-icon`

icon={icons.beer}
```

And here you just rename it to `bpk_${iconId.replace(/-/g, '_')}`

Reply:
Should I do the same with the color?